src/lib/form/validators.ts

```diff
@@ -192,6 +193,22 @@ export class Validators {
     return Validators.pattern(errorMessage, emailRegex);
   }
 
+  public static phoneNumber(errorMessage: string): ValidationRule {
+    const applies: ValidatorFn = (value: string) => {
+      if (Validators.required("").applies(value)) return false;
```

Comment:
Just thinking if this could be a little clearer here. I think I may have removed it but I had a function in this file of:

```typescript
function isEmptyString(value: string) {
// Do code to check if empty
}
```

which would make this message here clearer as on first reading (for me) it's not as clear as it could be what this line is doing. When I read it I would expect to read: if the value isn't empty then perform the validation.

Reply:
@NyWeb and I decided to remove `isEmptyString` in [this commit](https://github.com/mcagov/beacons-webapp/commit/66726eedab89516270b4f15721ca106f3800fced#diff-037a532fe1cb33971741cbd85fb74b66654e9ce218e618a507c35e93a88ba8ab).  

It worked when we only had a few simple validators / reasons to error, but with greater complexity it was harming readability because there started to be other reasons a validator wouldn't want to fire.  Having a set of functions to manage that was becoming unwieldy, especially as they would duplicate code already expressed in the `Validator` methods.

We decided to go for the [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html), calling the other Validator methods as predicates instead to effectively say "If `Validator X applies then don't apply this Validator`.

Especially useful for the hexId validator:

```typescript
  public static ukEncodedBeacon(errorMessage: string): ValidationRule {
    const applies: ValidatorFn = (value: string) => {
      if (Validators.required("").applies(value)) return false;
      if (Validators.isLength("", 15).applies(value)) return false;
      if (Validators.hexadecimalString("").applies(value)) return false;

      const ukCountryCodes = [232, 233, 234, 235];
      const beaconCountryCode = HexIdParser.countryCode(value);

      return !ukCountryCodes.includes(beaconCountryCode);
    };
```

Reply:
Another example here: 

![image](https://user-images.githubusercontent.com/54271433/110966098-d937d480-834c-11eb-814a-9dff76e2606b.png)

Reply:
Discussed with @matthew-a-carr .  This works for now, but we will likely want to shift the responsibility of _which validator should fire_ to the `FormManager` class.

Reply:
Raised issue #152

Reply:
#152